user_connections.go

```diff
@@ -0,0 +1,239 @@
+type ConnectedUsersPool struct {
+	handlers map[string]SockchatUserHandler
+	lock     sync.RWMutex
+}
+
+func NewUserManager() *ConnectedUsersPool {
+	manager := &ConnectedUsersPool{
+		handlers: make(map[string]SockchatUserHandler),
+	}
+	return manager
+}
+
+func (m *ConnectedUsersPool) GetHandler(nick string) (SockchatUserHandler, bool) {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	handler, ok := m.handlers[nick]
+	return handler, ok
+}
+
+func (m *ConnectedUsersPool) AddConnection(conn *SockChatWS) SockchatUserHandler {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		handler = NewUserHandler(conn)
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		m.handlers[conn.Nick] = handler
+	}
+	handler.AddConnection(conn)
+	return handler
+}
+
+func (m *ConnectedUsersPool) RemoveConnection(conn *SockChatWS) {
+	handler, ok := m.GetHandler(conn.Nick)
+	if !ok {
+		log.Printf("could not drop connection - no handler found for %s", conn.Nick)
+		return
+	}
+
+	if handler.getActiveConnectionsCount() > 1 {
+		handler.RemoveConnection(conn)
+	} else {
+		m.lock.Lock()
+		defer m.lock.Unlock()
+		delete(m.handlers, conn.Nick)
+	}
+}
+
+// UserHandler manages connections of a single connected user
+type UserHandler struct {
+	Nick                  string
+	connections           map[*SockChatWS]bool
+	createChannelRequests chan *CreateChannel
+	joinChannelRequests   chan *JoinChannel
+	leaveChannelRequests  chan *LeaveChannel
+	sendMessageRequests   chan *SendMessage
+	lock                  sync.RWMutex
+}
+
+type CreateChannel struct {
+	store       SockchatChannelStore
+	name        string
+	errCallback chan error
+}
+
+type JoinChannel struct {
+	store       SockchatChannelStore
+	channel     string
+	errCallback chan error
+}
+
+type LeaveChannel struct {
+	store       SockchatChannelStore
+	channel     string
+	errCallback chan error
+}
+
+type SendMessage struct {
+	store       SockchatChannelStore
+	channel     string
+	message     string
+	errCallback chan error
+}
+
+func NewUserHandler(conn *SockChatWS) *UserHandler {
+	handler := UserHandler{
+		Nick:                  conn.Nick,
+		connections:           make(map[*SockChatWS]bool),
+		createChannelRequests: make(chan *CreateChannel),
+		joinChannelRequests:   make(chan *JoinChannel),
+		leaveChannelRequests:  make(chan *LeaveChannel),
+		sendMessageRequests:   make(chan *SendMessage),
+	}
+	go handler.HandleRequests()
+	return &handler
+}
+
+func (u *UserHandler) HandleRequests() {
+	for {
+		select {
+		case req := <-u.createChannelRequests:
+			err := req.store.CreateChannel(req.name)
+			req.errCallback <- err
+
+		case req := <-u.joinChannelRequests:
+			err := req.store.AddUserToChannel(req.channel, u)
+			if err != nil {
+				req.errCallback <- err
+				continue
+			}
+			req.errCallback <- nil
+
+		case req := <-u.leaveChannelRequests:
+			err := req.store.RemoveUserFromChannel(req.channel, u)
+			if err != nil {
+				req.errCallback <- err
+				continue
+			}
+			req.errCallback <- nil
```

Comment:
```suggestion
			req.errCallback <- err
```
It can be simplified